src/corelibs/U2Gui/src/U2SavableWidget.cpp

```diff
@@ -188,10 +190,30 @@ MWMDIWindow* U2SavableWidget::getContextWindow() const {
     return contextWindow;
 }
 
+void U2SavableWidget::appendExcludeWidgets(const QStringList& excludeWidgetIds) {
+    for (const auto& excludeWgtId : qAsConst(excludeWidgetIds)) {
+        auto excludeWgt = getChildWidgetById(excludeWgtId);
+        SAFE_POINT(excludeWgt != nullptr, QString("Widget %1 not found").arg(excludeWgtId), );
+
+        excludeWidgets << excludeWgt;
+    }
+}
+
 QSet<QWidget*> U2SavableWidget::getCompoundChildren() const {
     return QSet<QWidget*>();
 }
 
+bool U2SavableWidget::isExcluded(const QString& childId) const {
+    auto wgt = getChildWidgetById(childId);
+    for (auto excludeWgt : qAsConst(excludeWidgets)) {
+        CHECK_CONTINUE(excludeWgt == wgt || excludeWgt->findChild<QWidget*>(childId) != nullptr);
```

Comment:
nit: use std::any_of

Reply:
q: why do you need this part: `excludeWgt->findChild<QWidget*>(childId)` ? I see that excluded widget can only come from ` auto excludeWgt = getChildWidgetById(excludeWgtId);`

Reply:
This part is required if you want to exclude a bunch of widgets, which all have a common ancestor. For example, here I excluded 2 widgets - region selector and annotation settings. For example, region selector contains several widgets - 2 labels, 2 line edits and popup menu:
![image](https://user-images.githubusercontent.com/26257527/231663022-945c35ee-c901-4006-bc79-e75f8d08ab31.png)

And annotation settings have more than 10 widgets. I do not need to enumerate them all - just to add their common parent.

Reply:
I see. Why do you call findChild twice for the same id? 
Once here `auto wgt = getChildWidgetById(childId);` and once here `excludeWgt->findChild<QWidget*>(childId)`
Are these 2 different hierarchies?

Reply:
I do not get this question, actually. I'll try to explain:

1. `excludeWgt == wgt` - check, that `excludeWgt` itself is not a widget, which should be exclude.
2. `excludeWgt->findChild<QWidget*>(childId) != nullptr` - check, that `excludeWgt` does not have children, which should be exclude

Reply:
I mean that your code iterates the whole `excludeWidgets` list subtree multiple times: once per each widget in the list.

This code is not good from the algorithmic POV:  all these widgets are within the same hierarchy, right? A single findChild should be enough. Next, for the child->parent relation you can check by parent: there is no need to iterate all children tree for it.

Next: check 'wgt' for nullptr. You solution will crash in user mode if the child does not exist (getChildWidgetById return nullptr).